src/Compilers/CSharp/Test/Semantic/Semantics/NameOfTests.cs

```diff
@@ -1168,5 +1168,66 @@ public void NameofTypeParameterInParameterInitializer()
 }";
             var compilation = CreateCompilationWithMscorlib45(source).VerifyDiagnostics();
         }
+
+        [Fact, WorkItem(10467, "https://github.com/dotnet/roslyn/issues/10467")]
+        public void NameofFixedBuffer()
+        {
+            var source =
+@"
+using System;
+unsafe struct Struct1
+{
+  public fixed char MessageType[50];
```

Comment:
What happens when I try and do `nameof` on `MessageType` from outside `Struct1` in a context that is safe?  Should that work?  We should check on that and add a test.